tests/tests_unit/test_schema.py

```diff
@@ -48,3 +53,33 @@ def test_render_schema(nebari_config):
     assert isinstance(nebari_config, schema.Main)
     assert nebari_config.project_name == f"pytest{nebari_config.provider.value}"
     assert nebari_config.namespace == "dev"
+
+
+@pytest.mark.parametrize(
+    "provider, exception",
+    [
+        (
+            "fake",
+            pytest.raises(
+                ValueError,
+                match="'fake' is not a valid enumeration member; permitted: local, existing, do, aws, gcp, azure",
+            ),
+        ),
+        ("aws", pytest.raises(ValidationError)),
+        ("gcp", pytest.raises(ValidationError)),
+        ("do", pytest.raises(ValidationError)),
+        ("azure", pytest.raises(ValidationError)),
```

Comment:
https://github.com/nebari-dev/nebari/blob/056b420059d5121898cfbf3db6c8da23241f333c/src/_nebari/stages/infrastructure/__init__.py#L528-L546

All cloud provider models have required fields so the current implementation does not work. I'll address this issue in another PR.

Reply:
Which don't work, the current implementation of the test or the initializations of the cloud provider objects?

Reply:
> Which don't work, the current implementation of the test or the initializations of the cloud provider objects?

The initializations of the cloud provider objects are not working. If we really want to initialize those objects here in the root_validator, we need to assign corresponding default values to all required fields.